Pod/Classes/SEGMoEngageIntegration.m

```diff
@@ -25,47 +25,79 @@ - (void)track:(SEGTrackPayload *)payload
 
 - (void)identify:(SEGIdentifyPayload *)payload
 {
-    NSDictionary *traits = payload.traits;
-
+    NSDictionary *moengagePayloadDict = [payload.traits copy];
+    
+    NSMutableDictionary *traits = [NSMutableDictionary dictionaryWithDictionary:moengagePayloadDict];
+    if(![traits.allKeys count]){
```

Comment:
I disagree on this. ! is a BOOL check. If (![traits.allKeys count]) will mean it will enter only if traits dictionary does not have any keys. An empty dict will have no keys. It will return if there are no keys. I am presuming that you can send a dict which is initialised and has no keys in it.